sdk/azidentity/interactive_browser_credential.go

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azidentity
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/pkg/browser"
+)
+
+// InteractiveBrowserCredentialOptions can be used when providing additional credential information, such as a client secret.
+// Also use these options to modify the default pipeline behavior through the TokenCredentialOptions.
+type InteractiveBrowserCredentialOptions struct {
```

Comment:
Presumably we want to add a `DefaultInteractiveBrowserCredentialOptions()` func like we did in `azcore`.

Reply:
I dont think so in this case. The defaults are applied automatically if nil is specified for the options. I could add a comment saying so.

Reply:
The idea is that if a user wants to configure one piece, they call `DefaultXXXOptions()` to get a pre-populated options struct with default values then change the part(s) they care about.

Reply:
So I've got a problem with this optional tenantID and optional clientID, because if we provide the defaults ("organizations" and the developer sign on client ID) if a user were to change only one, wouldnt it be broken?